### ISSUE TITLE
Is there a better term than "packaging ecosystem" that is more obvious to most readers?

### ISSUE THREAD

**Issue by joshuagl** [slsa-framework/slsa#538]
We are trying to define personas/implementers for the various classifications of SLSA requirements. One that we would appreciate wider feedback on is "packaging ecosystem", as defined below. 

* If you know of a better term please suggest it as a comment.
* If you agree that packaging ecosystem suits, please add a 👍 reaction (or any reaction other than 👎 or 😕).

> A packaging ecosystem is a set of conventions and tooling for package distribution. Every package has an ecosystem, whether it is formal or ad-hoc. Some ecosystems are formal, such as language distribution (e.g. [Python/PyPA](https://www.pypa.io/)), operating system distribution (e.g. [Debian/Apt](https://wiki.debian.org/DebianRepository/Format)), or artifact distribution (e.g. [OCI](https://github.com/opencontainers/distribution-spec)). Other ecosystems are informal, such as a convention used within a company. Even ad-hoc distribution of software, such as through a link on a website, is considered an “ecosystem”.

_**Note**: this issue somewhat duplicates #520, however this issue is created with the explicit purpose of inviting opinions from folks outside of SLSA._
Ref: https://github.com/slsa-framework/slsa/pull/536#discussion_r1030330808

**Comment by chair6** [comment]
Depends on how you define "package"? :) To me, "packaging ecosystem" feels a little specific, given the examples in the quoted definition above .. "software distribution ecosystem", perhaps?

**Comment by feelepxyz** [comment]
👋  We've grappled with this question at GitHub a fair bit and landed on "Package Ecosystem" as the least bad option. It's not perfect as there's some ambiguity but seems like the most common way to refer to these things.

Is there are reason for preferring `packaging` over `package`? 

Google results:
- "package ecosystem": 28,100
- "packaging ecosystem": 8,510

**Comment by joshuagl** [comment]
> 👋 We've grappled with this question at GitHub a fair bit and landed on "Package Ecosystem" as the least bad option. It's not perfect as there's some ambiguity but seems like the most common way to refer to these things.

This is very helpful context, thank you!

> Is there are reason for preferring `packaging` over `package`?

I don't think so, other than that "packaging ecosystem" is production oriented which aligns with SLSA's focus. Switching to "package ecosystem" should be a simple change. 🙂

**Comment by ciaracarey** [comment]
+1 for package ecosystem

**Comment by awaiken** [comment]
would like to understand how something so broad and potentially unique to individual organizations can be incorporated within the SLSA framework. "Other ecosystems are informal, such as a convention used within a company." How does "a" convention become a relevant packaging ecosystem?

**Comment by MarkLodato** [comment]
Any thoughts on shifting to the term "distribution" (and "distribution ecosystem") instead of "package" (and "package ecosystem")?

So something like:

![supply-chain-model-distribution](https://user-images.githubusercontent.com/58860/204317674-6a8f67a7-9690-4397-9eb1-1c1fc563f3c3.svg)

 I worry that "package" and "package ecosystem" puts too much emphasis on a formal packaging system like NPM or PyPI. The intention is that SLSA is not just done by each consumer individually, but rather that we shift the effort to some sort of "ecosystem" that takes care of it for most consumers. For example:

- A formal OSS package ecosystem, such as NPM, takes care of provenance distribution, expectations, and verifications so that users of the system are protected in an opt-in or opt-out basis, with very little effort.
- Within a company, central infrastructure takes care of provenance distribution, expectations, and verifications so that the hundreds or thousands of teams that use the infrastructure just need to opt-in or opt-out with little effort, without having to do a bunch of implementation work.

I'm wondering if "distribution" better conveys that idea?

**Comment by MarkLodato** [comment]
@joshuagl pointed out that "distribution" is even more overloaded than "package":

- Linux distribution (though does this actually align with our usage?)
- Python distribution == "package"

**Comment by sudo-bmitch** [comment]
Coming from the OCI side, I was leaning towards Artifact Repository, but that probably has it's own share of issues. I've got no objections to Packaging Ecosystem.

**Comment by Pierre-Gronau-ndaal** [comment]
my proposal: "package management system" like ...

**Comment by david-a-wheeler** [comment]
> my proposal: "package management system"

I don't agree. The problem is that many people will read "package management system" as *ONLY* the client-side tool for downloading & updating packages, aka a "package manager". But we expressly are *not* limiting it only to the client, but also the server side & related processes. It's better to use an unfamiliar term (that we get to define) than a familiar-looking term with a meaning that many might consider deceptive or at least misleading.

**Comment by Pierre-Gronau-ndaal** [comment]
quote from the NIST glossary:

An administrative tool or utility that facilitates the installation and maintenance of software on a given host, device or pool of centrally managed hosts, and the reporting of installed software attributes. May also be referred to as package manager, software manager, application manager, or app manager.

this is not a client only definition

**Comment by feelepxyz** [comment]
I still think "package ecosystem" is still the least bad option here. It's also used by the Open Source Vulnerability Schema (OpenSSF initiative): https://ossf.github.io/osv-schema/#affectedpackage-field

While [package manager](https://en.wikipedia.org/wiki/Package_manager) is a very common way to refer to these tools, to me it does imply that you're talking about the tool itself (e.g. CLI). This can get confusing in environments where there is no tool specifically to manage packages, for example GitHub Actions or Circle Orbs.

**Comment by MarkLodato** [comment]
OK, sounds like consensus around "package ecosystem." I'll send out a PR to fix "packaging" -> "package".
